New Climate entity logic

bob-tm opened the pull request:
- New climate logic, like original app
- Change entity names template. HA now can rename all device sensors with device rename.
- new device_class for duration and wifi sesnors
- Add retained flag for climate params (HA Restart unavailable bug)
- new sesnors:
  - binary_sensor_device_connected - device to grid connection status
  - sensor_active_setpoint - current temp to adjust from climate entity
  - sesnor_temperature_current - floor or room temp based on regulation type
  - sensor_temperature_target - target for climate entity

bob-tm committed:
Add files via upload

bob-tm committed:
Merge branch 'igor-podpalchenko:main' into main

bob-tm committed:
Update DeviRegHandler.java

db postfix removed

bob-tm committed:
Update DeviRegHandler.java

selectSetpointChannelByModeAndState fixed

bob-tm committed:
Update DeviRegHandler.java

bob-tm committed:
Update sensor_sys_run_time.json

bob-tm committed:
Update DeviRegHandler.java

CHANNEL_THERMOSTAT_PRESET fix

bob-tm committed:
Update DeviRegHandler.java

return formatDuration

bob-tm committed:
Update sensor_sys_run_time.json

bob-tm committed:
Update sensor_sys_run_time.json

bob-tm committed:
Update DanfossBindingConstants.java

bob-tm committed:
Update DeviRegHandler.java

currentSetpointCh

bob-tm committed:
Update DanfossBindingConstants.java

bob-tm committed:
Update DanfossBindingConstants.java

bob-tm committed:
Update DeviRegHandler.java

bob-tm committed:
Update climate_thermostat.json

bob-tm committed:
Update DanfossBindingConstants.java

bob-tm committed:
Update DeviRegHandler.java

bob-tm committed:
Add files via upload

bob-tm committed:
Proper device to grid connection handling

two new sensors
binary sensor: device_connected
sesnsor: device_connected text description

bob-tm committed:
Climate availability. New connected sensors

Climate availability_topic added
New sesnor: binary_sensor_device_connected.json 
New sesnor: sensor_device_connected.json

bob-tm committed:
FIx HA Restart Bug

Mqtt Retained flag set to some important messages

stlk commented on auto-discovery-templates/autogen-sensors.py:
```diff
@@ -26,6 +26,18 @@
+    "on_time_30_days": "s",
+    "on_time_7_days": "s",
+    "on_time_total": "s",
+    "on_time_30_days_kwh":"kwh",
```
HA expects the unit to be `kWh` otherwise they won't be visible in the Energy section.

bob-tm replied:
Tnx. Updated in my fork.

stlk reviewed (commented):
These changes are just what I've been looking for! I'm usually using thermostats on manual setting and that works very good. I've got only one note regarding `unit_of_measurement` HA expects.

bob-tm committed:
Fix Negative Power Report 

Fixed negative sys_info_output_power report (-3200 watt in my case)
Fixed kwh report (if sys_info_output_power > 0 do not work with negative power)

bob-tm committed:
kwh to kWh

bob-tm committed:
Update to kWh sensor_on_time_30_days_kwh.json

bob-tm committed:
Update to kWh sensor_on_time_7_days_kwh.json

bob-tm committed:
Update to kWh sensor_on_time_total_kwh.json

igor-podpalchenko reviewed (approved):
Merging to main

igor-podpalchenko commented:
Check the new release. Pull the latest docker.
